src/formatting/closures.rs

```diff
@@ -124,19 +124,19 @@ fn needs_block(block: &ast::Block, prefix: &str, context: &RewriteContext<'_>) -
 // TODO: The expressions "veto"ed here should align with expressions that are permitted on multiple
 // lines in rewrite_closure_expr#allow_multi_line. Consider refactoring to avoid this disparity.
 fn veto_block(e: &ast::Expr) -> bool {
-    match e.kind {
+    matches!(
+        e.kind,
         ast::ExprKind::Call(..)
-        | ast::ExprKind::Binary(..)
-        | ast::ExprKind::Cast(..)
-        | ast::ExprKind::Type(..)
-        | ast::ExprKind::Assign(..)
-        | ast::ExprKind::AssignOp(..)
-        | ast::ExprKind::Field(..)
-        | ast::ExprKind::Index(..)
-        | ast::ExprKind::Range(..)
-        | ast::ExprKind::Try(..) => true,
-        _ => false,
-    }
+            | ast::ExprKind::Binary(..)
+            | ast::ExprKind::Cast(..)
+            | ast::ExprKind::Type(..)
+            | ast::ExprKind::Assign(..)
+            | ast::ExprKind::AssignOp(..)
+            | ast::ExprKind::Field(..)
+            | ast::ExprKind::Index(..)
+            | ast::ExprKind::Range(..)
+            | ast::ExprKind::Try(..)
+    )
```

Comment:
I take it this is coming from `match_like_matches_macro`? TBH, I find the original match usage to be more readable. Using the macro and the resulting the extra indentation just looks odd

Reply:
D'oh sorry, see you have the lint names in the commit message